plaso/containers/artifacts.py

```diff
@@ -219,6 +219,176 @@ def IsEquivalent(self, other):
     return False
 
 
+class PathArtifact(ArtifactAttributeContainer):
+  """Path artifact attribute container.
+
+  Attributes:
+    data_stream (str): name of a data stream.
+    path_segment_separator (str): path segment separator.
+    path_segments (list[str]): path segments.
+  """
+  CONTAINER_TYPE = 'path'
+
+  def __init__(self, data_stream=None, path=None, path_segment_separator='/'):
+    """Initializes a path artifact.
+
+    Args:
+      data_stream (Optional[str]): name of a data stream.
+      path (Optional[str]): a path.
+      path_segment_separator (Optional[str]): path segment separator.
+    """
+    super(PathArtifact, self).__init__()
+    self.data_stream = data_stream
+    self.path_segment_separator = path_segment_separator
+    self.path_segments = self._SplitPath(path, path_segment_separator)
+
+  def __eq__(self, other):
+    """Determines if the path is equal to other.
+
+    Args:
+      other (str): path to compare against.
```

Comment:
Shouldn't we allow comparing PathArtifact objects with eachother?

Reply:
we should, I was planning to add this as a later stage.